### PR TITLE
test(common): coverage for desktop tab navigation

### PR DESCRIPTION
**Pull request by CuriousCorrelation**
This adds test coverage for the tab navigation methods added in FE-907
 the native tab keyboard shortcuts implementation.

 Closes FE-909

 The recently implemented desktop tab keyboard shortcuts added new tab
 navigation methods to the `TabService` class (`goToNextTab()`,
 `goToPreviousTab()`, `goToFirstTab()`, `goToLastTab()`,
 `goToTabByIndex()`, and `reopenClosedTab()`). These methods lacked
 corresponding test coverage.

 ### Test coverage added

 Tab Navigation Tests:
 - `goToNextTab()` - Verifies cycling through tabs in forward direction with wrap-around behavior
 - `goToPreviousTab()` - Verifies cycling through tabs in reverse direction with wrap-around behavior
 - `goToFirstTab()` - Verifies navigation to the leftmost tab regardless of current position
 - `goToLastTab()` - Verifies navigation to the rightmost tab regardless of current position
 - `goToTabByIndex()` - Verifies direct navigation to tabs by 1-based index position

 Edge Case Handling:
 - Invalid tab indices should not change active tab
 - Single tab scenarios where navigation methods should maintain current tab
 - Boundary conditions for wrap-around navigation behavior

 Recently Closed Tabs Tests (this is a WIP feature):
 - `reopenClosedTab()` - Verifies LIFO restoration of closed tabs with correct positioning
 - Closed tabs history maintenance with proper ordering
 - Tab restoration at original position in tab list
 - Handling empty closed tabs stack

 ### Implementation details

 The tests use the existing infra.

 ### Notes to reviewers

 The "Recently Closed Tabs" functionality is marked as WIP in the
 original implementation but tests are included just for completeness.

 The test suite uses `@ts-ignore` (or rather `@typescript-eslint/no-unused-vars`) for 
 unused variables in some scenarios where creating multiple tabs is necessary 
 for readability but not all tab references are used in assertions.

### CODE REVIEW

File: packages/hoppscotch-common/src/services/tab/__tests__/tab.service.spec.ts [hunk 1]
```diff
@@ -132,4 +132,213 @@ describe("TabService", () => {
...
+      expect(service.getActiveTab()?.id).toEqual(tab3.id)
+    })
+
+    it("should navigate to first tab", () => {
+      const container = new TestContainer()
+      const service = container.bind(MockTabService)
+
+      // Unused variable that improves readability.
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const tab2 = service.createNewTab({ request: "second request" })
```
[the remark this answers is not in the document]
Comment (CuriousCorrelation) [reply packages/hoppscotch-common/src/services/tab/__tests__/tab.service.spec.ts:183]: Sure but unfortunately it still gets flagged by `lsp`:
```
tab.service.spec.ts:181:13 lsp-flycheck-error-unnecessary 6133
    '_tab2' is declared but its value is never read. (lsp)
``` 
so having more cross-editor cross-tool approach; that is most likely to work everywhere, is generally preferable.

File: packages/hoppscotch-common/src/services/tab/__tests__/tab.service.spec.ts [hunk 2]
```diff
@@ -132,4 +132,213 @@ describe("TabService", () => {
...
+      expect(service.getActiveTab()?.id).toEqual("test")
+    })
+
+    it("should navigate to last tab", () => {
+      const container = new TestContainer()
+      const service = container.bind(MockTabService)
+
+      // Unused variable that improves readability.
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const tab2 = service.createNewTab({ request: "second request" })
```
[the remark this answers is not in the document]
Comment (CuriousCorrelation) [reply packages/hoppscotch-common/src/services/tab/__tests__/tab.service.spec.ts:199]: Same as above.

**Review by jamesgeorge007 — APPROVED** [review]
